src/ConfidentialLedger/docs/Update-AzConfidentialLedger.md

```diff
@@ -0,0 +1,321 @@
+---
+external help file:
+Module Name: Az.ConfidentialLedger
+online version: https://docs.microsoft.com/powershell/module/az.confidentialledger/update-azconfidentialledger
+schema: 2.0.0
+---
+
+# Update-AzConfidentialLedger
+
+## SYNOPSIS
+Updates properties of Confidential Ledger
+
+## SYNTAX
+
+### UpdateExpanded (Default)
+```
+Update-AzConfidentialLedger -Name <String> -ResourceGroupName <String> [-SubscriptionId <String>]
+ [-AadBasedSecurityPrincipal <IAadBasedSecurityPrincipal[]>]
+ [-CertBasedSecurityPrincipal <ICertBasedSecurityPrincipal[]>] [-LedgerType <LedgerType>] [-Location <String>]
```

Comment:
My mistake. I just find there is a parameter Location. I believe Location can't be updated. Is it true? Have you tested each parameter locally. We always meet the case like that. Some immutable parameter shows in update- command.

Reply:
How can we hide it - is it a setting in the API swagger or the readme?

Reply:
To clarify, it is a required field for the API call. We mark `location` with `x-mis-mutability: ["create", "read"]` in the [Swagger](https://github.com/Azure/azure-rest-api-specs/blob/06f446933347cf235bf85abe1afdbee49615bf9c/specification/confidentialledger/resource-manager/Microsoft.ConfidentialLedger/preview/2021-05-13-preview/confidentialledger.json#L415), which [indicates that the property can not be updated after creation](https://azure.github.io/autorest/extensions/#x-ms-mutability). How canI resolve this?

Reply:
Here is an instruction: https://github.com/Azure/autorest.powershell/blob/main/docs/directives.md#parameter-hiding. I will let xiaogang know this issue.

Reply:
Just tested - our RP returns the following if no location is provided `"{\"error\":{\"code\":\"ResourcePatchValidateFailed\",\"message\":\"The resource validation failed.\"}}"`. I saw in the example that you can provide a default value via the module; however, I suspect that if a different value than the region in which the resource is located is passed in the request, the RP will also return an error. Thus, we cannot have the PowerShell module set a default value either.

Reply:
So I guess the location must be provided and it's one of identifying information. If so, I think we may need to mark the location as mandatory parameter so that customer have to provide it.

Reply:
Is there a directive for that, or is it sufficient to just edit the documentation?

Reply:
It's insufficient. We need to change Location from optional to required in command syntax level. Customization 
can help us. There are 3 steps:
- Copy Update-AzConfidentialLedger.ps1 from exports folder to custom folder
- Hide generated Update-AzConfidentialLedger by directive
- Edit custom/Update-AzConfidentialLedger.ps1, set Location as mandatory
https://github.com/Azure/azure-powershell/blob/fd1c2fa8e72c911c8ac94d958cd89a62162b1a80/src/Kusto/custom/Update-AzKustoDataConnection.ps1#L179-L183

Reply:
Got it, thanks! I've now hidden the auto-generated code and use the custom cmdlet.